crates/bevy_ecs/examples/change_detection.rs

```diff
@@ -0,0 +1,73 @@
+use bevy_ecs::prelude::*;
+use std::ops::Deref;
+
+#[derive(Debug)]
+struct Counter {
+    pub value: i32,
+}
+
+fn main() {
+    // Create a world
+    let mut world = World::new();
+
+    // Add the counter resource
+    world.insert_resource(Counter { value: 0 });
+
+    // Create a schedule and a stage
+    let mut schedule = Schedule::default();
+    let mut update = SystemStage::parallel();
+
+    // Add systems sending and receiving events
+    update.add_system(increase_counter.system().label("increase"));
+    update.add_system(print_counter_when_changed.system().after("increase"));
+    update.add_system(manipulate_entities.system().label("mutate"));
+    update.add_system(print_changed_entities.system().after("mutate"));
+    schedule.add_stage("update", update);
+
+    for iteration in 1..=10 {
+        println!("Simulating frame {}/10", iteration);
+        schedule.run_once(&mut world);
+    }
+}
+
+fn manipulate_entities(
+    mut commands: Commands,
+    last_spawned_entity: Query<Entity, Without<i32>>,
+    mut entities: Query<&mut i32>,
+) {
+    if rand::random::<f32>() > 0.5 {
+        commands.spawn();
+    }
+    for mut value in entities.iter_mut() {
+        *value += 1;
+    }
+    for entity in last_spawned_entity.iter() {
```

Comment:
This doesn't behave as expected. Entities from commands.spawn() won't show up in the `last_spawned_entity` query until the next frame. This example also doesn't really make sense functionally. In this case people should just call `commands.spawn().insert(0)`.

* I think we should use non-integer components to ground these examples in some "reality".
* I think we should only use programing patterns that we would encourage in real life. This particular pattern seems "wrong" to me.